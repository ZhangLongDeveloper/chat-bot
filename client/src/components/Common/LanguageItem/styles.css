.language-container {
  max-width: 150px;
  display: inline-block;
  margin: 10px;
  text-align: center;
}
.language-container .country-flag {
  width: 100%;
  border: 1px solid black;
  box-shadow: 1px 1px 10px gray;
}
.language-container .name {
  font-size: 18px;
  font-weight: 500;
}
.language-container .level {
  font-size: 14px;
  color: gray;
}

@media (max-width: 750px) {
  .language-container .name {
    font-size: 14px;
    font-weight: 500;
  }
  .language-container .level {
    font-size: 12px;
    color: gray;
  }
}

@media (max-width: 500px) {
  .language-container {
    max-width: 100px;
  }
  .language-container .level {
    font-size: 10px;
  }
}