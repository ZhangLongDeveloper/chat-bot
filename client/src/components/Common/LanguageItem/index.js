import React from 'react';
import PropTypes from 'prop-types';

import './styles.css';

const LanguageItem = ({ image, name, level }) => (
  <div className="language-container">
    <img src={image} alt="flag" className="country-flag" />
    <div className="name">{name}</div>
    <div className="level">{level}</div>
  </div>
);

LanguageItem.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  level: PropTypes.string.isRequired,
};

export default LanguageItem;
