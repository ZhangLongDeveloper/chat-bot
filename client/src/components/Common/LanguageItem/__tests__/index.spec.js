import React from 'react';
import { create } from 'react-test-renderer';

import LanguageItem from '../../LanguageItem';

const mockComponent = props => (
  <LanguageItem {...props} />
);

describe('components/Common/<LanguageItem />', () => {
  it('should render component', () => {
    const props = {
      data: {
        name: 'name',
        image: 'image',
        level: 'level',
      },
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
