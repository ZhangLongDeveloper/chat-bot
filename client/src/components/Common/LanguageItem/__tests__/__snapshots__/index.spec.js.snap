// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Common/<LanguageItem /> should render component 1`] = `
<div
  className="language-container col s6 m3"
>
  <img
    alt="flag"
    className="country-flag"
  />
  <div
    className="name"
  />
  <div
    className="level"
  />
</div>
`;
